components/layout_2020/flow/inline.rs

```diff
@@ -393,6 +391,118 @@ impl<'box_tree> PartialInlineBoxFragment<'box_tree> {
+    };
+    start_corner += &relative_adjustement(
+        &atomic.style,
+        ifc.containing_block.inline_size,
+        ifc.containing_block.block_size,
+    );
+
+    let fragment = match atomic.as_replaced() {
+        Ok(replaced) => {
+            // FIXME: implement https://drafts.csswg.org/css2/visudet.html#inline-replaced-width
+            let inline_size = Length::zero();
+            let block_size = Length::zero();
+            let containing_block_for_children = ContainingBlock {
+                inline_size,
+                block_size: LengthOrAuto::LengthPercentage(block_size),
+                mode: atomic.style.writing_mode(),
+            };
+            // https://drafts.csswg.org/css-writing-modes/#orthogonal-flows
+            assert_eq!(
+                ifc.containing_block.mode, containing_block_for_children.mode,
+                "Mixed writing modes are not supported yet"
+            );
+            let independent_layout = replaced.layout(&atomic.style, &containing_block_for_children);
+            let content_rect = Rect {
+                start_corner,
+                size: Vec2 {
+                    block: independent_layout.content_block_size,
+                    inline: inline_size,
+                },
+            };
+            BoxFragment {
+                style: atomic.style.clone(),
+                children: independent_layout.fragments,
+                content_rect,
+                padding,
+                border,
+                margin,
+                block_margins_collapsed_with_children: CollapsedBlockMargins::zero(),
+            }
+        },
+        Err(non_replaced) => {
+            let box_size = atomic.style.box_size();
+            let inline_size = box_size.inline.percentage_relative_to(cbis).auto_is(|| {
+                let available_size =
+                    cbis - padding.inline_sum() - border.inline_sum() - margin.inline_sum();
+                shrink_to_fit(&atomic.inline_content_sizes, available_size)
+            });
+            let block_size = box_size
+                .block
+                .maybe_percentage_relative_to(ifc.containing_block.block_size.non_auto());
+            let containing_block_for_children = ContainingBlock {
+                inline_size,
+                block_size,
+                mode: atomic.style.writing_mode(),
+            };
+            assert_eq!(
+                ifc.containing_block.mode, containing_block_for_children.mode,
+                "Mixed writing modes are not supported yet"
+            );
+            // FIXME is this correct?
+            let dummy_tree_rank = 0;
+            // FIXME: Do we need to call `adjust_static_positions` somewhere near here?
+            let independent_layout = non_replaced.layout(
+                layout_context,
+                &containing_block_for_children,
+                dummy_tree_rank,
+                ifc.absolutely_positioned_fragments,
+            );
```

Comment:
A dummy tree rank should only be used if this is an absolute containing block (i.e. a containing block in which absolutely positioned descendants should be placed relatively to).

Reply:
That’s what I suspected. But I think line breaking makes it tricky to computed the correct rank, especially when we’ll add support for going “back” through an atomic inline-level to find the last wrap opportunity after overflowing a line.